docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,202 @@
+# Resource Adoption 
+
+# The Problem
+
+New ACK users typically have existing AWS resources in their accounts. These resources were not originally created by an ACK service controller. While ACK can create new resources, getting ACK service controllers to take over management of an existing resource is not yet supported.
+
+The current method for migrating management of resources over to ACK requires users to store the current state of their applications, delete the resources, and re-create them using ACK. This process introduces places for bugs, loss of the state of existing deployments and downtime in production workloads.
+
+ACK aims to provide the ability to “adopt” existing resources into your Kubernetes environment. Given the user can provide an IAM role which has access to existing resource, and that ACK operators exist for each resource’s associated service, the ACK controller will be able to determine the current specification and status of the AWS resource and reconcile said resource as if it were created natively in k8s.
+
+## Use Cases
+
+1. Migrating existing architecture from another platform (CFN, CDK, Terraform, etc.)
+    1. Resources exist but have never been managed by K8s - no spec files
+3. Migrating architecture across K8s environments
+    1. Resource are being managed by another K8s environment
+
+# Proposed Implementation
+
+## Overview
+
+The proposed solution will provide an alternate workflow by which to adopt resources through a separate set of manifests. A new custom resource definition, the `AdoptedResource` CRD, provides a signal to ACK to adopt an existing resource. 
+The `AdoptedResource` CRD references an existing AWS resource (either through ARN, name or other unique identifier) and a target Kubernetes `GroupVersionKind`. Each ACK controller will reconcile all adopted resources that match any of the types it controls (for the service it manages).
+The manager will describe the referenced AWS resource, typically through a `Describe*` SDK call, and use the returned values to compile a new custom resource - with filled spec and status fields. The manager will then apply this custom resource to the cluster using the metadata applied in the `AdoptedResource` spec.
+
+## `AdoptedResource` Specification
+
+The `AdoptedResource` custom resource definition will allow for specification of an AWS resource unique identifier (in the `aws` field) and a target `kubernetes` field - containing Kubernetes `GroupVersionKind` and `metadata` sub-fields.
+
+The `aws` resource unique identifier field allows identification through at most one of:
+
+* `arn` - An [ARN](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) (Amazon Resource Name)
+* `name` - A resource name (a string value)
+* `id` - A resource ID (a string value)
```

Comment:
I have been chatting with @surajkota about the object relationship feature stuff and it's pretty clear to me that we will want to have some sort of `ackcorev1alpha1.AWSIdentifiers` struct type that contains the above oneOf {name,id,arn}. Please consider making such a struct as part of your implementation so that it may be re-used in things like a concrete object relationship type, for example:

In `apis/core/v1alpha1/identifiers.go`:

```go
type AWSIdentifiers struct {
    // ARN is the AWS Resource Name for the resource. It is a globally
    // unique identifier.
    ARN *AWSResourceName `json:"arn"`
    // Name is a user-supplied string identifier for the resource. It may
    // or may not be globally unique, depending on the type of resource.
    Name *string `json:"name"`
    // ID is an identifier that has been generated by the AWS service for
    // the resource. ID fields are *usually*, but not always, globally unique.
    // Sometimes ID fields are numeric, other times they are strings.
    ID *string `json:"id"`
}
```

and then in, for example, `services/sagemaker/apis/v1alpha1/relations.go`, we might find:

```go
import (
    ackcorev1alpha1 "github.com/aws/aws-controllers-k8s/apis/core/v1alpha1"
)

// ModelReference is a pointer to a Model resource in the Sagemaker API
// and/or the Kubernetes API.
type ModelReference struct {
    // AWS indicates the AWS external identifiers for the Model.
    // If nil, the `Kubernetes` field must not be nil.
    AWS *ackv1alpha1.AWSIdentifiers `json:"aws,omitempty"`
    // Kubernetes indicates the Kubernetes internal identifiers for
    // the Model CR that references the AWS resource. If nil, the
    // Model CR has not yet been created by an ACK service
    // controller and the `AWS` field above should be non-nil.
    Kubernetes *ackcorev1alpha1.PartialObjectMeta `json:"aws,omitempty"
}
```